### PR TITLE
Lead facilitator teachercon registration take 2

### PR DESCRIPTION
**Pull request by mehalshah**
https://trello.com/c/Re7TzECu/465-lead-facilitators-can-register-for-teachercon

### CODE REVIEW

File: apps/src/code-studio/pd/teachercon1819_registration/Teachercon1819FormComponent.jsx [hunk 1]
```diff
@@ -16,4 +17,10 @@ export default class Teachercon1819FormComponent extends LabeledFormComponent {
   isPartnerApplication() {
     return this.props.applicationType === PARTNER;
   }
+
+  isLeadFacilitatorApplication() {
+    console.log(this.props.applicationType);
```
Comment (clareconstantine) [inline apps/src/code-studio/pd/teachercon1819_registration/Teachercon1819FormComponent.jsx]: remove these `console.log`s

File: dashboard/app/mailers/pd/teachercon1819_registration_mailer.rb [hunk 2]
```diff
@@ -12,9 +12,16 @@ def teacher(registration)
 
   def facilitator(registration)
     @registration = registration
+    email = registration.pd_application_id? ? registration.pd_application.user.email : registration.user.email
```
Comment (clareconstantine) [inline dashboard/app/mailers/pd/teachercon1819_registration_mailer.rb]: if all registrations have a user, could we just use `registration.user.email` here?

File: dashboard/app/views/pd/teachercon1819_registration_mailer/facilitator.html.haml [hunk 3]
```diff
@@ -1,21 +1,27 @@
 %p
   Hi
-  = @registration.pd_application.first_name + ','
-
+  - if @registration.pd_application
+    = @registration.pd_application.first_name + ','
+  - else
+    = @registration.sanitize_form_data_hash[:preferred_first_name] + ','
 %p
```
Comment (clareconstantine) [inline dashboard/app/views/pd/teachercon1819_registration_mailer/facilitator.html.haml:7]: Can we use the preferred first name from the registration for both of these cases, and avoid the conditional?

File: dashboard/app/views/pd/teachercon1819_registration_mailer/lead_facilitator.html.haml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+- if @registration.accepted?
+  %p
+    We look forward to seeing you this summer! You will receive more information about
+    travel approximately six weeks before TeacherCon. In the meantime, please visit
+    - info_link = "code.org/educate/teachercon/#{@registration.pd_application.workshop.location_city.downcase}"
+    = link_to(info_link, "https://#{info_link}")
+    for more information.
+- if @registration.declined?
+  %p
```
Comment (clareconstantine) [inline dashboard/app/views/pd/teachercon1819_registration_mailer/lead_facilitator.html.haml]: is there a case where the form might be neither accepted nor declined? if it's always one or the other, I would use `else` to communicate that.

File: dashboard/app/views/pd/teachercon1819_registration_mailer/facilitator.html.haml (hunk 3 above)
Comment (mehalshah) [reply dashboard/app/views/pd/teachercon1819_registration_mailer/facilitator.html.haml:7]: We can't. `first_name` is a function that returns `hash[:preferred_first_name] || hash[:first_name]`

**Pull request merged**